tests/vector/test_info_format_strategies.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from gym.error import InvalidInfoFormat
+from gym.vector.utils import (
+    BraxVecEnvInfoStrategy,
+    ClassicVecEnvInfoStrategy,
+    get_info_strategy,
+)
+
+
+@pytest.mark.parametrize(("num_envs"), [3])
+def test_classic_vec_env_info_strategy(num_envs):
+    infos = ClassicVecEnvInfoStrategy(num_envs)
+    for i in range(num_envs):
+        info = {"example_info": i}
+        infos.add_info(info, i)
+
```

Comment:
Could you add an extra line only a single environment info is provided. i.e. `infos.add_info({'env 1 info': 1}, 1)`

Reply:
This condition should not happen imho; in a vectorized environment the standard behavior was looping through the envs and appending the info to a list. The ClassicInfoStrategy indeed implements `env_num: int` as parameter just to adhere the interface but it actually don't use it:  

for a quick reference:

``` python
observations, infos = [], []
for i, (env, action) in enumerate(zip(self.envs, self._actions)):
  observation, self._rewards[i], self._dones[i], info = env.step(action)
  if self._dones[i]:
      info["terminal_observation"] = observation
      observation = env.reset()
  observations.append(observation)
  infos.append(info)
```

so for example by doing
```infos.add_info({'env 1 info': 1}, 1)``` 
on this 
``` [{"example_info": 0}, {"example_info": 1}, {"example_info": 2}]``` 

we will obtain
 ``` [{"example_info": 0}, {"example_info": 1}, {"example_info": 2}, {'env 1 info': 1}]```

Reply:
Ohh I get what you mean, ignore my comment. I forgot about the info data being updated after it being passed to the info strategy. Do we want a check in the `add_info` that the data does not already exist or do we assume people will not be stupid? (like me)?